src/components/audio-player-view/audio-player-view.tsx

```diff
@@ -117,8 +117,10 @@ const AudioPlayerView = Event.withEventManager(
           const [showPluginsMenuOverlay, setShowPluginsMenuOverlay] = useState(false);
           const [wasPlaying, setWasPlaying] = useState(false);
 
+
+          const availablePlugins = usePluginsManager(player.getService('AudioPluginsManager'))
```

Comment:
i don't think that a component should access the service directly, i think the access to service should be in a separate file.
for example, you can create the manager in audio-player.tsx and pass it to the component via props or useContext.

Reply:
i don’t see any issue with this, the UI components access player instance all over the place, any way you can not pass it from outside, since the plugin class doesn’t have control over the components since it is a custom preset injected form outside it is not injecting it as the rest of the plugins